src/dotty/tools/dotc/transform/FunctionalInterfaces.scala

```diff
@@ -63,7 +63,12 @@ class FunctionalInterfaces extends MiniPhaseTransform {
         val m = tree.meth.tpe.widen.asInstanceOf[MethodType]
 
         if (shouldSpecialize(m)) {
-          val interfaceName = (functionName ++ m.paramTypes.length.toString).specializedFor(m.resultType, m.paramTypes)
+          val functionSymbol = tree.tpe.widenDealias.classSymbol
+          val names = ctx.atPhase(ctx.erasurePhase.prev) {
```

Comment:
Why `erasurePhase.prev`? Note that at `erasurePhase` we are still *before* erasure. At `erasurePhase.prev` we are one phase ahead (i.e. at the beginning of `resolveSuper`. Is that really where we want to be? If yes, we need at least a comment.